### PR TITLE
[Bug Fix] Depop Charm Pet and Detach Debuffs on Evacuate

### PR DESCRIPTION
**Pull request by fryguy503**
This will clear charm from charm pets and deteach debuffs to prevent some social aggro issues and exploitable conditions with charming and pulling a mob across the zone with no aggro concerns.

### CODE REVIEW

File: zone/entity.cpp [hunk 1]
```diff
@@ -3536,20 +3536,26 @@ void EntityList::HalveAggro(Mob *who)
 }
 
 //removes "targ" from all hate lists, including feigned, in the zone
-void EntityList::ClearAggro(Mob* targ)
-{
+void EntityList::ClearAggro(Mob* targ, bool clear_caster_id) {
```
Comment (Kinglykrab) [inline zone/entity.cpp:3539]: Usually we put `{` on new line for methods.

**Review by Kinglykrab — COMMENTED** [review]
Is this Live-like, what issue occurs here exactly?

**Comment by fryguy503** [comment]
I am honestly unsure, I can rule is off for safety (I haven't played on live in eons)

**Comment by Kinglykrab** [comment]
> I am honestly unsure, I can rule is off for safety (I haven't played on live in eons)

What kind of issues could this cause if it went in as-is? Just removes effects from NPC on zone?

**Review by Kinglykrab — APPROVED** [review]

**Pull request merged**